src/three-components/Renderer.js

```diff
@@ -61,10 +63,18 @@ export default class Renderer extends EventDispatcher {
       context: this.context,
     });
     this.renderer.autoClear = false;
-    this.renderer.setPixelRatio(DPR);
     this.renderer.gammaOutput = true;
     this.renderer.gammaFactor = GAMMA_FACTOR;
 
+    this[$updateDpr] = debounce(() => {
+      this.dpr = resolveDpr();
+      this.renderer.setPixelRatio(this.dpr);
+    }, UPDATE_DPR_DEBOUNCE_MS);
+
+    this.dpr = resolveDpr();
+    this.renderer.setPixelRatio(this.dpr);
+    window.addEventListener('resize', this[$updateDpr]);
```

Comment:
Is this fired on 'zoom' as well?

Reply:
It's not. I think I will remove this for now, as measuring DPR is no longer as expensive as it was at the beginning of this change. Instead. we can measure it in real time.